docs/porting/libraries.md

```diff
@@ -1,5 +1,311 @@
+
+### Staying on .NET Framework until portability issues are resolved
+
+This approach may be the best if you prefer to have code that compiles during the entire process.  The approach is as follows:
+
+1. Run API Port on a project.
+2. Address issues by using different APIs which are portable.
+3. Keep note of any areas where you can't use a direct alternative.
+4. Repeat steps 1-3 for all projects you're porting until you're confident each is ready to be copied over into a .NET Core project.
+5. Copy the code into a new .NET Core projects.
+6. Work out any issues that you've kept note of.
+
+This careful approach is more structured than simply working out compiler errors, but it is still relatively code-focused and has the benefit of always having code that can compile.  The way you resolve certain issues that couldn't be addressed by just using another API can vary greatly.  You may find that you need to develop a more comprehensive plan for certain projects, which is covered as the next approach.
+
+### Developing a comprehensive plan of attack
+
+This approach may be best for larger and more complex projects, where restructuring of code or rewriting certain areas may be necessary to support .NET Core.  The approach is as follows:
+
+1. Run API Port on a project.
+2. Understand where in your code each non-portable type is being used and how that affects overall portability.
+
+   a.  Understand the nature of those types.  Are they small in number, but used frequently?  Are they large in number, but used infrequently?  Is their use concentrated, or is it spread throughout your code?
+   
+   b. Is it easy to isolate code that isn't portable so you can deal with it more easily?
+   
+   c. Would you need to refactor your code?
+   
+   d. For those types which aren't portable, are there alternative APIs that accomplish the same task?  For example, if you're using the `WebClient` class, you may be able to use the `HttpClient` class instead.
+   
+   e. Are there different portable APIs you can use to accomplish a task, even if it's not a drop-in replacement?  For example, if you're using `XmlSchema` to help parse XML but you don't require XML schema discovery, you could use Linq to XML APIs and hand-parse the data.
+
+3. If you have assemblies that are difficult to port, is it worth leaving them on .NET Framework for now?
+
+   a. You may have some functionality in your library that's incompatible with .NET Core because it relies too heavily on .NET Framework- or Windows-specific functionality.  Is it worth leaving that functionality behind for now and releasing a .NET Core version of your library with less features for the time being?
+   
+   b. Would a refactor help here?
+   
+4. Is it reasonable to write some code a .NET Framework API was doing yourself?
```

Comment:
"Is it reasonable to write your own implementation of an unavailable .NET Framework API?"